tests/unit/util/test_auto_rule_corpus_tester.py

```diff
@@ -0,0 +1,437 @@
+    @mock.patch("logprep.util.auto_rule_corpus_tester.parse_json")
+    def test_run_logs_json_decoding_error(
+        self,
+        mock_parse_json,
+        mock_exit,
+        tmp_path,
+        corpus_tester,
+    ):
+        test_data = {
+            "input": {"winlog": {"event_id": "2222", "event_data": {"Test1": 1, "Test2": 2}}},
+            "expected_output": {
+                "winlog": {"event_id": "2222", "event_data": "<IGNORE_VALUE>"},
+                "test_normalized": {"test": {"field1": 1, "field2": 2}},
+            },
+            "expected_extra_output": [],
+        }
+        expected_prints = [
+            "FAILED",
+            "Json-Error decoding file rule_auto_corpus_test_out.json:",
+            "Json-Error decoding file rule_auto_corpus_test_out_extra.json:",
+            "Some Error: line 1 column 1 (char 0)",
+            "Success rate: 0.00%",
+        ]
+        prepare_corpus_tester(corpus_tester, tmp_path, test_data)
+        mock_parse_json.side_effect = JSONDecodeError("Some Error", "in doc", 0)
+        with corpus_tester.console.capture() as capture:
+            corpus_tester.run()
+        console_output = capture.get()
+        for expected_print in expected_prints:
+            assert expected_print in console_output
+        mock_exit.assert_called_with(1)
+
+    def test_run_raises_if_case_misses_input_file(self, tmp_path, corpus_tester):
+        expected_output_data_path = tmp_path / "rule_auto_corpus_test_out.json"
+        expected_output_data_path.write_text("not important")
+        corpus_tester.input_test_data_path = tmp_path
+        with pytest.raises(ValueError, match="is missing an input file."):
+            corpus_tester.run()
+
+    @mock.patch("logprep.util.auto_rule_corpus_tester.sys.exit")
+    def test_run_skips_test_if_expected_output_is_missing(self, mock_exit, tmp_path, corpus_tester):
+        test_data = {
+            "input": {"winlog": {"event_id": "2222", "event_data": {"Test1": 1, "Test2": 2}}},
+            "expected_output": {
+                "winlog": {"event_id": "2222", "event_data": "<IGNORE_VALUE>"},
+                "test_normalized": {"test": {"field1": 1, "field2": 2}},
+            },
+            "expected_extra_output": [],
+        }
+        expected_prints = [
+            "SKIPPED",
+            "no expected output given",
+            "Total test cases: 1",
+            "Success rate: 100.00%",
+        ]
+        prepare_corpus_tester(corpus_tester, tmp_path, test_data)
+        os.remove(tmp_path / "rule_auto_corpus_test_out.json")
+        with corpus_tester.console.capture() as capture:
+            corpus_tester.run()
+        console_output = capture.get()
+        for expected_print in expected_prints:
+            assert expected_print in console_output
+        mock_exit.assert_called_with(0)
+
+    def test_patch_config_rewrites_input_output_and_process_count(self, corpus_tester, tmp_path):
```

Comment:
the amount of assertions makes me feel it could be 3 tests. Tests should be short and readable.

Reply:
not relevant anymore after many refactorings.